website/src/components/FlaggableElement.tsx

```diff
@@ -0,0 +1,220 @@
+import {
+  Button,
+  Checkbox,
+  Flex,
+  Popover,
+  PopoverAnchor,
+  PopoverArrow,
+  PopoverBody,
+  PopoverCloseButton,
+  PopoverContent,
+  PopoverTrigger,
+  Slider,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderTrack,
+  Spacer,
+  useBoolean,
+} from "@chakra-ui/react";
+import { FlagIcon, QuestionMarkCircleIcon } from "@heroicons/react/20/solid";
+
+import poster from "src/lib/poster";
+import useSWRMutation from "swr/mutation";
+import { useState } from "react";
+
+export const FlaggableElement = (props) => {
+  const [isEditing, setIsEditing] = useBoolean();
+  const { trigger, isMutating } = useSWRMutation("/api/v1/text_labels", poster, {
+    onSuccess: () => {
+      setIsEditing.off;
+    },
+  });
+
+  const submitResponse = (label_data) => {
+    trigger(label_data);
+  };
+  const [checkboxValues, setCheckboxValues] = useState(new Array(TEXT_LABEL_FLAGS.length).fill(false));
+  const [sliderValues, setSliderValues] = useState(new Array(TEXT_LABEL_FLAGS.length).fill(1));
+
+  const fetchData = () => {
```

Comment:
small detail, does this need to be a separate function? It seems to only be called when calling submitResponse.  We could simplify this by putting it all in submitResponse

Reply:
thank you! I've added the same and ran precommit and npm run build. No issues now!

Reply:
![image](https://user-images.githubusercontent.com/15769643/210196708-44a628d3-14e8-4f34-832e-7060f1f03dcc.png)